src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -0,0 +1,180 @@
+import _isEmpty from 'lodash/isEmpty';
+import _isObject from 'lodash/isObject';
+import _map from 'lodash/map';
+import LegModel from './LegModel';
+import { routeStringFormatHelper } from '../../navigationLibrary/Route/routeStringFormatHelper';
+
+/**
+ *
+ * This class should always be instantiated from an `AircraftInstanceModel` and
+ * always instantiated with some form of a `spawnPatternModel`.
+ *
+ * @class Fms
+ */
+export default class Fms {
+    /**
+     * @constructor
+     * @param aircraftInitProps {object}
+     * @param initialRunwayAssignment {string}
+     * @param navigationLibrary {NavigationLibrary}
+     */
+    constructor(aircraftInitProps, initialRunwayAssignment, navigationLibrary) {
+        if (!_isObject(aircraftInitProps) || _isEmpty(aircraftInitProps)) {
+            throw new TypeError('Invalid aircraftInitProps passed to Fms');
+        }
+
+        /**
+         *
+         *
+         * @property _navigationLibrary
+         * @type {NavigationLibrary}
+         * @private
+         */
+        this._navigationLibrary = navigationLibrary;
+
+        /**
+         *
+         *
+         * @property _runway
+         * @type {string}
+         * @private
+         */
+        this._runway = initialRunwayAssignment;
+
+        /**
+         *
+         *
+         * @property legCollection
+         * @type {array}
+         * @default []
+         */
+        this.legCollection = [];
+
+        /**
+         *
+         *
+         * @property category
+         * @type {string}
+         * @default ''
+         */
+        this.category = '';
```

Comment:
the only time we need both `runway` and `category` is when we are building a `LegModel` for a `procedureRoute`.  Perhaps those can both go away?

I'm using category here the same way we use it with `SpawnPatternModels`.  Though true an aircraft could be all three of those, it is only one at any given time.  But, as I mentioned above, this property is only important when building a `procedureRoute`.